ci/rust-version.sh

```diff
@@ -46,7 +46,7 @@ export rust_nightly_docker_image=solanalabs/rust-nightly:"$nightly_version"
     if ! cargo +"$toolchain" -V > /dev/null; then
       echo "$0: Missing toolchain? Installing...: $toolchain" >&2
       rustup install "$toolchain"
-      rustup +"$toolchain" componnent add clippy rustfmt
```

Comment:
oops. there's no `set -e` here...? prefer separate pr...

Reply:
I think you are looking at an incremental change in the PR, rather than the PR as a whole.
Here is the commit in this PR that adds this line: [solana-labs/solana@`07602dc` (#31276)](https://github.com/solana-labs/solana/pull/31276/commits/07602dcbe3f40b1be7dc7207d244440280cdeb3f)

I've added this to see if the CI would make more progress before the images would be available.
The change adds this whole line
```
rustup +"$toolchain" component add clippy rustfmt
```
I do not fully understand how the CI is structured, so, let me remove this change altogether.

Overall, it seems that CI runs rustup with the "minimal" profile.
At the same time, expecting clippy and rustfmt.  Both are not part of the "minumal" profile.
The command has no effect if the component was already installed by the preceding `rustup install "$toolchain"`.

Reply:
Here is the component addition as a separate PR: https://github.com/solana-labs/solana/pull/31280

Reply:
Hm... Seems like this change is indeed necessary: https://buildkite.com/solana-labs/solana/builds/94223#01879d48-2e7a-4c7b-8367-869da1aa8102

I do not understand it, but let me bring the component addition change back into this PR.

Reply:
Ah, it did not use the updated image, it seems.

The CI log shows:

```
++exec docker run --workdir /solana --volume
  /var/lib/buildkite-agent/builds/ci37/solana-labs/solana:/solana --rm --volume
  /var/lib/buildkite-agent:/home --env RUSTC_WRAPPER=/home/.cargo/bin/sccache
  --env AWS_ACCESS_KEY_ID --env AWS_SECRET_ACCESS_KEY --env SCCACHE_BUCKET --env
  SCCACHE_REGION --env SCCACHE_S3_KEY_PREFIX --env HOME=/home --env
  CARGO_HOME=/home/.cargo --security-opt seccomp=unconfined --user 997:997 --env
  BUILDKITE --env BUILDKITE_AGENT_ACCESS_TOKEN --env BUILDKITE_JOB_ID --env CI
  --env CI_BRANCH --env CI_BASE_BRANCH --env CI_TAG --env CI_BUILD_ID --env
  CI_COMMIT --env CI_JOB_ID --env CI_PULL_REQUEST --env CI_REPO_SLUG --env
  CRATES_IO_TOKEN -e CODECOV_ENV -e CODECOV_TOKEN -e CODECOV_URL -e CODECOV_SLUG
  -e VCS_COMMIT_ID -e VCS_BRANCH_NAME -e VCS_PULL_REQUEST -e VCS_SLUG -e VCS_TAG
  -e CI_BUILD_URL -e CI_BUILD_ID -e CI_JOB_ID -e CI -e BUILDKITE -e
  BUILDKITE_BRANCH -e BUILDKITE_BUILD_NUMBER -e BUILDKITE_JOB_ID -e
  BUILDKITE_BUILD_URL -e BUILDKITE_PROJECT_SLUG -e BUILDKITE_COMMIT -t
  solanalabs/rust-nightly:2023-01-22 ci/test-checks.sh
--


ci/test-checks.sh: Missing toolchain? Installing...: 1.68.2
```

So it is using `solanalabs/rust-nightly`, rather than `solanalabs/rust:latest`.

Reply:
our `check` and `coverage` tests are using nightly. btw, I'm still trying to fix sccache so that we can upgrade our nightly version. I prefer this PR only upgrade the stable rust version.

Reply:
I have removed the `rustup components` command.
But CI is failing now.

Reply:
ah.. sorry. one sec.

Reply:
just retried the pipeline. it's running 🪖